src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Schemas/ControlToken.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.PowerFx.LanguageServerProtocol.Protocol;
+
+namespace Microsoft.PowerFx.LanguageServerProtocol.Schemas
+{
+    internal class ControlToken
```

Comment:
>ControlToken [](http://example.com/codeflow?start=19&length=12)

Can we add the ctor and mark fields readonly?

Reply:
I can add one, but we do not use this right now anywhere.

Reply:
The reason is, we want field read-only. right now you don't need it becuase you are directly assigning fields, which in the long run are prone to bugs. 
Meaning sometimes people may change the value of fields at a wired point without realizing it.
Basically, we always prefer read-only fields unless its really need to be not one.

Reply:
Fair enough, making the changes.